### PR TITLE
added post merge hook using husky

### LINKED ISSUE #1108: Adding post-merge functionality for the husky to improve developer experience.
**Describe the bug**
Currently, we dont have any post-merge functionality in husky... Sometimes, when some new dependencies are added to package.json and user git pulls the files to his local system, he gets errors and some of the required node modules will be missing....To prevent this , we can add post-merge functionality so that everytime a user uses 'git pull', if the remote repo had some changes into package.json, 'npm install' should automatically run

**To Reproduce**

1.Update your remote repo and then use git pull
2.You will see this error if the remote had some changes in pakage.json

**Expected behavior**
no errors should be there and the developing experience should be good

**Actual behavior**
might get errors sometime and developers may get stuck

**Screenshots**
not required

### PR DESCRIPTION
**Pull request by anurag0006**
**What kind of change does this PR introduce?**
a bugfix

**Issue Number:**
#1108 
Fixes #

**Did you add tests for your changes?**
no

**Snapshots/Videos:**
not required

**If relevant, did you update the documentation?**
Yes

**Summary**
This PR will improve user experience. Now whenever some made a PR with some changes to the pakage.json, if any other person will pull those files to his local system, he will automatically get the latest dependencies and hence closes #1108 

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

**Comment by anurag0006** [comment]
@palisadoes, In this PR i had to add a new file for 'post-merge'. Its saying that the workflow requires approval from the maintainer...Can you please approve

**Comment by palisadoes** [comment]
- Please update the first comment so that the issue autocloses. The PR_GUIDELINES.md file has details.
- For this PR accessing the restricted files is OK

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -236,9 +236,23 @@ To fix formatting issues in code use this command:-
 
 We are using the package `Husky` to run a pre-commit hook which automatically runs these checks each time you make a commit and also fixes some of the issues. This way you don't have to run them manually each time.
```
Comment (xoldyckk) [inline CONTRIBUTING.md:237]: Change the details to just mention that we're running hooks that run according to different git workflows.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -236,9 +236,23 @@ To fix formatting issues in code use this command:-
...
 
+
 If you don't want these pre-commit checks running on each commit, you can manually opt out of it using the `--no-verify` flag with your commit message as shown:-
 
         git commit -m "commit message" --no-verify
+
+
+<br/>
+
+We are also using `Husky` to run a post-commit hook which will automatically run "npm install" only if there is any change made to pakage.json file so that the developer has all the required dependencies when pulling files from remote.
```
Comment (xoldyckk) [inline CONTRIBUTING.md:247]: This isn't a post-commit hook, it's a post-pull hook, mention it as such

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -236,9 +236,23 @@ To fix formatting issues in code use this command:-
 
 We are using the package `Husky` to run a pre-commit hook which automatically runs these checks each time you make a commit and also fixes some of the issues. This way you don't have to run them manually each time.
 
+
 If you don't want these pre-commit checks running on each commit, you can manually opt out of it using the `--no-verify` flag with your commit message as shown:-
```
Comment (xoldyckk) [inline CONTRIBUTING.md:240]: Extract these two hooks into their own sub headings. Also add those headings in table of contents.

**Review by xoldyckk — COMMENTED** [review]
When writing documentation for a project follow a certain standard. Make sure your intent is clear, easy to understand, structured and easy to navigate. This isn't just writing side comments for some code file, a certain standard is expected here because it is read by many contributors.

File: CONTRIBUTING.md (hunk 3 above)
Comment (anurag0006) [reply CONTRIBUTING.md:240]: Yeah, sorry for the mistake, i got a bit confused and post-commit and post-pull so mistakenly wrote it....i will make subheading for both of them and try to make the PR again

**Comment by anurag0006** [comment]
@xoldyckk , i made the required changes, please have a look

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -234,20 +234,27 @@ To fix formatting issues in code use this command:-
 
 <br/>
 
-We are using the package `Husky` to run a pre-commit hook which automatically runs these checks each time you make a commit and also fixes some of the issues. This way you don't have to run them manually each time.
+We are using the package `Husky` to run git hooks that run according to different git workflows.
+
+<br/>
+
+#### pre-commit hook
+we run a pre-commit hook which automatically runs these checks each time you make a commit and also fixes some of the issues. This way you don't have to run them manually each time.
```
Comment (xoldyckk) [inline CONTRIBUTING.md:242]: `runs code quality checks`

**Review by xoldyckk — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -234,11 +234,32 @@ To fix formatting issues in code use this command:-
...
 If you don't want these pre-commit checks running on each commit, you can manually opt out of it using the `--no-verify` flag with your commit message as shown:-
 
         git commit -m "commit message" --no-verify
+
+<br/>
+
+
+#### post-merge hook
+
+we are also running a post-merge(post-pull) hook which will automatically run "npm install" only if there is any change made to pakage.json file so that the developer has all the required dependencies when pulling files from remote.
```
Comment (xoldyckk) [inline CONTRIBUTING.md:254]: Capitalize `we`

**Comment by palisadoes** [comment]
@anurag0006 Let me know when you've made these changes so that we can merge.

**Comment by anurag0006** [comment]
@palisadoes  all the required changes have been made...Please merge this PR

**Review by xoldyckk — APPROVED** [review]

**Comment by xoldyckk** [comment]
@anurag0006 Try to resolve the conversations when you fix the stuff asked in them

**Review by palisadoes — APPROVED** [review]

**Pull request merged**